waku/waku_relay/protocol.nim

```diff
@@ -179,22 +183,33 @@ method stop*(w: WakuRelay) {.async.} =
   debug "stop"
   await procCall GossipSub(w).stop()
 
-# rejects messages that are not WakuMessage
-proc validator(pubsubTopic: string, message: messages.Message): Future[ValidationResult] {.async.} =
-  # can be optimized by checking if the message is a WakuMessage without allocating memory
-  # see nim-libp2p protobuf library
-  let msg = WakuMessage.decode(message.data)
-  if msg.isOk():
-    return ValidationResult.Accept
-  return ValidationResult.Reject
-
 proc isSubscribed*(w: WakuRelay, topic: PubsubTopic): bool =
   GossipSub(w).topics.hasKey(topic)
 
 iterator subscribedTopics*(w: WakuRelay): lent PubsubTopic =
   for topic in GossipSub(w).topics.keys():
     yield topic
 
+proc generateOrderedValidator*(w: WakuRelay): auto {.gcsafe.} =
```

Comment:
The 'Ordered' in this case refers to first validating the basic _Waku Message_ encoding, and then, the rest?

Reply:
Sort of, we describe here 1 single validator that gossipsub calls back to - and we control the order of